hw/drivers/lp5523/include/lp5523/lp5523.h

```diff
@@ -0,0 +1,426 @@
+
+#define LP5523_INS_MUX_LD_START(addr) \
+    (0x9e00 | \
+    (addr & 0x7f))
+
+#define LP5523_INS_MUX_MAP_START(addr) \
+    (0x9c00 | \
+    (addr & 0x7f))
+
+#define LP5523_INS_MUX_LD_END(addr) \
+    (0x9c80 | \
+    (addr & 0x7f))
+
+#define LP5523_INS_MUX_SEL(sel) \
+    (0x9d00 | \
+    (sel & 0x7f))
+
+#define LP5523_INS_MUX_CLR() \
+    LP5523_INS_MUX_SEL(0)
+
+#define LP5523_INS_MUX_MAP_NEXT() \
+    (0x9d80)
+
+#define LP5523_INS_MUX_MAP_PREV() \
+    (0x9dc0)
+
+#define LP5523_INS_MUX_LD_NEXT() \
+    (0x9d81)
+
+#define LP5523_INS_MUX_LD_PREV() \
+    (0x9dc1)
+
+#define LP5523_INS_MUX_LD_ADDR(addr) \
+    (0x9f00 | \
+    (addr & 0x7f))
+
+#define LP5523_INS_MUX_MAP_ADDR(addr) \
+    (0x9f80 | \
+    (addr & 0x7f))
+
+#define LP5523_INS_RST() \
+    (0x0000)
+
+#define LP5523_INS_BRANCH_IM(loop_count, step_number) \
+    (0xa000 | \
+    ((loop_count & 0x3f) << 7) | \
+    (step_number & 0x7f))
+
+#define LP5523_INS_BRANCH(step_number, loop_count) \
+    (0x8600 | \
+    ((step_number & 0x7f) << 2) | \
+    (loop_count & 0x03))
+
+#define LP5523_INS_INT() \
+    (0xc400)
+
+#define LP5523_INS_END(interrupt, reset) \
+    (0xc000 | \
+    ((interrupt != 0) ? (1 << 12) : 0) | \
+    ((reset != 0) ? (1 << 11) : 0))
+
+#define LP5523_INS_TRIGGER(wait_external, wait_engines, send_external, \
+    send_engines) \
+    (0xe000 | \
+    ((wait_external != 0) ? (1 << 12) : 0) | \
+    ((wait_engines & 0x07) << 7) | \
+    ((send_external != 0) ? (1 << 6) : 0) | \
+    ((send_engines & 0x07) << 1))
+
+#define LP5523_INS_JNE(skip, variable1, variable2) \
+    (0x8800 | \
+    ((skip & 0x1f) << 4) | \
+    ((variable1 & 0x03) << 2) | \
+    (variable2 & 0x03))
+
+#define LP5523_INS_JL(skip, variable1, variable2) \
+    (0x8a00 | \
+    ((skip & 0x1f) << 4) | \
+    ((variable1 & 0x03) << 2) | \
+    (variable2 & 0x03))
+
+#define LP5523_INS_JGE(skip, variable1, variable2) \
+    (0x8c00 | \
+    ((skip & 0x1f) << 4) | \
+    ((variable1 & 0x03) << 2) | \
+    (variable2 & 0x03))
+
+#define LP5523_INS_JE(skip, variable1, variable2) \
+    (0x8e00 | \
+    ((skip & 0x1f) << 4) | \
+    ((variable1 & 0x03) << 2) | \
+    (variable2 & 0x03))
+
+#define LP5523_INS_LD(target_variable, value) \
+    (0x9000 | \
+    ((target_variable & 0x03) << 10) | \
+    (value & 0xff))
+
+#define LP5523_INS_ADD_IM(target_variable, value) \
+    (0x9100 | \
+    ((target_variable & 0x03) << 10) | \
+    (value & 0xff))
+
+#define LP5523_INS_ADD(target_variable, variable1, variable2) \
+    (0x9300 | \
+    ((target_variable & 0x03) << 10) | \
+    ((variable1 & 0x03) << 2) | \
+    (variable2 & 0x03))
+
+#define LP5523_INS_SUB_IM(target_variable, value) \
+    (0x9200 | \
+    ((target_variable & 0x03) << 10) | \
+    (value & 0xff))
+
+#define LP5523_INS_SUB(target_variable, variable1, variable2) \
+    (0x9310 | \
+    ((target_variable & 0x03) << 10) | \
+    ((variable1 & 0x03) << 2) | \
+    (variable2 & 0x03))
```

Comment:
For the macros above, can you please add #defines for individual bits and register addresses. It would make these macros easier to understand. Also, can you please add some comments for what the macros do. 

For example: 
```
#define LP5523_INS_ADD_IM(target_variable, value) \
    (0x9100 | \
    ((target_variable & 0x03) << 10) | \
    (value & 0xff)) 
```

would be 

```
#define LP5523_INS_ADD_IM(target_variable, value) \
    ( LP5523_OP_ADD_IM | \
    ((target_variable & LP5523_VAR_TARGET) << LP5523_VAR_TARGET_POS) | \
    (value & 0xff))
```

So, you would basically have defines for variables: LP5523_VAR_1, LP5523_VAR_2,  LP5523_VAR_1_POS and LP5523_VAR_2_POS. 

Also, for LED map instructions you could use 0x9c00 as a bitmap : LP5523_INS_LED_MAP, so that the class of the instruction is known. I know generalising for all of them is not possible but LED_MAP and DATA_ARITH seem generalizable.